### PR TITLE
Reorganize and edit the section "Restrict cross namespace resource associations"

### PR DESCRIPTION
**Pull request by alaudazzi**
The PR:
- Consolidates the two original sections ([Restrict cross namespace resource associations](https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-restrict-cross-namespace-associations.html) and [Enabling cross-namespace association restrictions](https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s_enabling_cross_namespace_association_restrictions.html)) into one single page
- Proposes a minor reorg of the introductory text
- Adds numbered steps

### CODE REVIEW

File: docs/operating-eck/restrict-cross-namespace-associations.asciidoc [hunk 1]
```diff
@@ -5,24 +5,22 @@ link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}.html[View
...
+This section describes how to restrict associations that can be created between resources managed by ECK.
 
-This section describes how you can restrict the associations that can be created between resources managed by ECK.
+When using the `elasticsearchRef` field to establish a trust relationship between Elasticsearch, Kibana, or APM Server resources, by default the association is allowed as long as both resources are deployed to namespaces managed by that particular ECK instance. The association is successful even if the user does not have access to one of the namespaces, or has only access to Kibana resources and not Elasticsearch resources.
 
-== Enabling cross-namespace association restrictions
+The restriction of cross-namespace associations is disabled by default. Once enabled, it only enforces an access control for resources deployed across two different namespaces. You can still create associations between resources deployed in a same namespace.
```
Comment (charith-elastic) [inline docs/operating-eck/restrict-cross-namespace-associations.asciidoc:14]:
```suggestion
The enforcement of access control rules for cross-namespace associations is disabled by default. Once enabled, it only enforces access control for resources deployed across two different namespaces. Associations between resources deployed in the same namespace are not affected.
```

File: docs/operating-eck/restrict-cross-namespace-associations.asciidoc [hunk 2]
```diff
@@ -80,9 +78,9 @@ spec:
   serviceAccountName: associated-resource-sa
 ----
 
-In the above example, `associated-resource` can be of any `Kind` that requires an association to be created (for example `Kibana` or the `APMServer`).
+In the above example, `associated-resource` can be of any `Kind` that requires an association to be created, for example `Kibana` or the `APMServer`.
```
Comment (charith-elastic) [inline docs/operating-eck/restrict-cross-namespace-associations.asciidoc:81]:
```suggestion
In the above example, `associated-resource` can be of any `Kind` that requires an association to be created, for example `Kibana` or `ApmServer`.
```

File: docs/operating-eck/restrict-cross-namespace-associations.asciidoc [hunk 3]
```diff
@@ -5,24 +5,22 @@ link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}.html[View
...
-= Restrict cross namespace resource associations
+= Restrict cross-namespace resource associations
 
-When using the `elasticsearchRef` field to establish a trust relationship between Elasticsearch and a Kibana or an APM Server resource, the default behaviour is to allow association as long as both resources are deployed to namespaces managed by that particular ECK instance. Association succeeds even if the user does not have access to one of the namespaces, or only has access to Kibana resources and not Elasticsearch resources.
+This section describes how to restrict associations that can be created between resources managed by ECK.
 
-This section describes how you can restrict the associations that can be created between resources managed by ECK.
+When using the `elasticsearchRef` field to establish a trust relationship between Elasticsearch, Kibana, or APM Server resources, by default the association is allowed as long as both resources are deployed to namespaces managed by that particular ECK instance. The association is successful even if the user does not have access to one of the namespaces, or has only access to Kibana resources and not Elasticsearch resources.
```
Comment (charith-elastic) [inline docs/operating-eck/restrict-cross-namespace-associations.asciidoc:12]:
```suggestion
When using the `elasticsearchRef` field to establish a connection to Elasticsearch from Kibana, APM Server, Enterprise Search, or Beats resources, by default the association is allowed as long as both resources are deployed to namespaces managed by that particular ECK instance. The association will succeed even if the user creating the association does not have access to one of the namespaces or the Elasticsearch resource.
```

File: docs/operating-eck/restrict-cross-namespace-associations.asciidoc [hunk 4]
```diff
@@ -5,24 +5,22 @@ link:https://www.elastic.co/guide/en/cloud-on-k8s/master/k8s-{page_id}.html[View
...
 
-== Enabling cross-namespace association restrictions
+The restriction of cross-namespace associations is disabled by default. Once enabled, it only enforces an access control for resources deployed across two different namespaces. You can still create associations between resources deployed in a same namespace.
 
-This feature is disabled by default. To enable it, start the operator with the `--enforce-rbac-on-refs` flag.
-
-NOTE: This feature only enforces an access control for resources deployed across two different namespaces. You can still create associations between resources deployed in a same namespace.
-
-Once enabled, associations are allowed as long as the `ServiceAccount` specified in the associated resource can execute HTTP `GET` requests against the referenced Elasticsearch object.
+Associations are allowed as long as the `ServiceAccount` specified in the associated resource can execute HTTP `GET` requests against the referenced Elasticsearch object.
```
Comment (charith-elastic) [inline docs/operating-eck/restrict-cross-namespace-associations.asciidoc:16]:
```suggestion
Associations are allowed as long as the `ServiceAccount` used by the associated resource can execute HTTP `GET` requests against the referenced Elasticsearch object.
```

File: docs/operating-eck/restrict-cross-namespace-associations.asciidoc [hunk 5]
```diff
@@ -80,9 +78,9 @@ spec:
   serviceAccountName: associated-resource-sa
 ----
 
-In the above example, `associated-resource` can be of any `Kind` that requires an association to be created (for example `Kibana` or the `APMServer`).
+In the above example, `associated-resource` can be of any `Kind` that requires an association to be created, for example `Kibana` or the `APMServer`.
 You can find https://github.com/elastic/cloud-on-k8s/blob/master/config/recipes/associations-rbac/apm_es_kibana_rbac.yaml[a complete example in the ECK GitHub repository].
 
-NOTE: If the `serviceAccountName` is not set, then ECK uses the `ServiceAccount` called `default`.
+NOTE: If the `serviceAccountName` is not set, ECK uses the `ServiceAccount` called `default`.
```
Comment (charith-elastic) [inline docs/operating-eck/restrict-cross-namespace-associations.asciidoc:84]:
```suggestion
NOTE: If the `serviceAccountName` is not set, ECK uses the default service account assigned to the pod by the link:https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#service-account-admission-controller[Service Account Admission Controller].
```

**Comment by alaudazzi** [comment]
Thank you Charith! I integrated your feedback.

**Pull request merged**